coordinator/core/core.go

```diff
@@ -319,12 +320,20 @@ func (c *Core) getStatus(ctx context.Context) (int, string, error) {
 	return int(c.state), status, nil
 }
 
-func (c *Core) generateSecrets(ctx context.Context, secrets map[string]Secret) (map[string]Secret, error) {
+func (c *Core) generateSecrets(ctx context.Context, secrets map[string]Secret, id uuid.UUID) (map[string]Secret, error) {
 	// Create a new map so we do not overwrite the entries in the manifest
 	newSecrets := make(map[string]Secret)
 
 	// Generate secrets
 	for name, secret := range secrets {
+
+		// Skip secrets from wrong context
+		if secret.Shared && id != uuid.Nil {
+			continue
+		} else if !secret.Shared && id == uuid.Nil {
+			continue
+		}
```

Comment:
```suggestion
		if secret.Shared != (id == uuid.Nil) {
			continue
		}
```
I find this easier to read, but if you disagree, you can keep it as is.

Reply:
That's fine, I'll take it